socorro/external/postgresql/products.py

```diff
@@ -14,11 +18,131 @@
 logger = logging.getLogger("webapi")
 
 
-class Products(PostgreSQLBase):
+class SmartDate(object):
+
+    def clean(self, value):
+        if any(itertools.imap(value.startswith, ('>=', '<='))):
+            operator_ = value[:2]
+            value = value[2:]
+        elif any(itertools.imap(value.startswith, ('=', '>', '<'))):
+            operator_ = value[:1]
+            value = value[1:]
+        else:
+            operator_ = '='
+        return (operator_, string_to_datetime(value).date())
+
+
+class ProductVersions(PostgreSQLBase):
+
+    def get(self, **kwargs):
+        filters = [
+            ("version", None, [str]),
+            ("product", None, [str]),
+            ("featured", None, bool),
+            ("start_date", None, SmartDate()),
+            ("end_date", None, SmartDate()),
+            ("active", None, bool),
+            ("is_rapid_beta", None, bool),
+        ]
+        params = external_common.parse_arguments(filters, kwargs, modern=True)
+        where = []
+        sql_params = {}
+        for param, value in params.items():
+            if value is None:
+                continue
+            param = {
+                'product': 'product_name',
+                'version': 'version_string',
+                'featured': 'is_featured',
+            }.get(param, param)
+
+            if param == 'active':
+                # This is a convenient exception. It makes it possible
+                # to query for only productversions that are NOT sunset
+                # without having to do any particular date arithmetic.
+                param = 'end_date'
+                operator_ = value and '>' or '<='
```

Comment:
``active``, to me, means that the end date is greater than or equal to today's date, not strictly greater. If the end date is today, it is still active until we hit midnight. 

So I would do ``value and '>=' or '<'``.